src/server/cluster/cluster_family.cc

```diff
@@ -736,8 +738,18 @@ void ClusterFamily::RemoveOutgoingMigrations(const std::vector<MigrationInfo>& m
     auto it = std::find_if(outgoing_migration_jobs_.begin(), outgoing_migration_jobs_.end(),
                            [&m](const auto& om) { return m == om->GetMigrationInfo(); });
     DCHECK(it != outgoing_migration_jobs_.end());
+    DCHECK(it->get() != nullptr);
+    OutgoingMigration& migration = *it->get();
+    if (migration.GetState() != MigrationState::C_FINISHED) {
```

Comment:
we can cancel even finished migration

Reply:
state can be changed in the different thread

Reply:
> we can cancel even finished migration

Not really, it requires unregistering from db slice, for which we may have not registered to (like in the case of the first config assignment)

> state can be changed in the different thread

Do you mean that there could be a race here?

Reply:
Both checks are done under `migration_mu_` - is there another lock I should use?

Reply:
migration_mu_ is used to create migrations, but state is changed inside migration

Reply:
There doesn't seem to be a lock which locks the inner fiber's state changes at all, right?